src/ser/mod.rs

```diff
@@ -163,10 +163,13 @@ impl PrettyConfig {
 
     /// Configures whether to emit struct names.
     ///
+    /// Note that when using the `explicit_struct_names` extension, this method will use an OR operation on the extension and the [`PrettyConfig`] option. See also [`Extensions::EXPLICIT_STRUCT_NAMES`] for the extension equivalent.
+    ///
     /// Default: `false`
     #[must_use]
     pub fn struct_names(mut self, struct_names: bool) -> Self {
-        self.struct_names = struct_names;
+        self.struct_names =
```

Comment:
I think this is the wrong place to do the OR-ing - what about the struct_names helper function in the Serializer (see https://github.com/ron-rs/ron/blob/00a65cdd9e01cfc54066387069f5218b09a74378/src/ser/mod.rs#L642) which should now look at the pretty config, pretty config extensions, and default extensions (is there already a union of those somewhere?)